manim/mobject/number_line.py

```diff
@@ -50,6 +94,53 @@ def __init__(
         x_max=config["frame_x_radius"],
         **kwargs
     ):
+        """
+
+        Parameters
+        ----------
+        color : :class:`~.Colors`, optional
+            the color for the number line, by default LIGHT_GREY
+        unit_size : :class:`int`, optional
```

Comment:
it should be optional only when the default value is `None`, otherwise no.

Reply:
Sorry, am not sure what is it that you are suggesting here.

Reply:
Essentially since the function header is like
https://github.com/ksachdeva/manim/blob/7f4839e4a96c08592088d0dbeaac5efcb75344b5/manim/mobject/number_line.py#L69
where the default value is 1, it should be changed to 
```suggestion
        unit_size : :class:`int`
```

Reply:
ah, now I see what you mean.

I use a tool to generate the docstrings and then fill them. This tool is essentially an addon to the VSCode to generate python documentation. 

In other words, this "int, optional" was generated by the tool and not written me. 

I would think that the tool is doing its job as per some policy (PEP?) but then a tool can have a bug as well so here is my request:

Would you be able to point at some doc/pep/policy where it is specified that one should **not** have `optional` if the value is **not** `None`? 

The reason I am asking you for that -:

I do not find any issue by specifying `optional` because to me it indeed indicates that it is an optional field with "some" default value. I do not care whether that value is None or not. That said, this is subjective but then the tool does it as well. Hence  I am requesting you to provide the reference which tells us what is (or should be) the right convention here.

Regards
Kapil

Reply:
https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
I believe this usage (and the other down further below are fine):

> 
> https://docs.python.org/3/library/typing.html#typing.Optional

Also, taken from that article:
`Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. For example:
`

And even in our own wiki docs (right above this header, see `def my_function(...., test=45):` https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP)#types

`Optional[.....]`, and `optional` are different.

Reply:
@jsonvillanueva somehow I had not seen the documentation guidelines. At least I will now keep an eye on it to be compliant. Thanks for the link.

From what I am seeing in the documentation of `my_function` for `test=45`; the `optional` is specified in the docstrings. Very much like what the addon in VSCode did.

Unless I have totally misread or misunderstood something here, there is no change required from my side.

Reply:
I figured you had already seen it since the documentation has been for the most part accurate, but nice addon! That might be very useful in more quickly documenting the rest of Manim.

As an aside, I think the only thing left is to change the path specification for numpy, see the example given at the bottom of this section: https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP)#path-specifications

Reply:
Sorry for the confusion, I think according to https://github.com/numpy/numpydoc/blob/master/doc/format.rst#sections which manim follow it allowed for having keyword arguments as optional but isn't allowed in typing though.

Reply:
makes sense. let me fix it

Reply:
@jsonvillanueva I think specifying as `~.Colors` should work. The link in docs is clickable https://manimce--1064.org.readthedocs.build/en/1064/reference/manim.mobject.number_line.NumberLine.html#manim.mobject.number_line.NumberLine

Reply:
Yes, it used to be `Colors` and not clickable prior to my recommendation/suggestion: https://github.com/ManimCommunity/manim/pull/1064#discussion_r584241033

@ksachdeva which addon is this? I wonder if it's configurable to automatically add this `~.` prefix to files from the local Manim library.

Reply:
yeah, `numpy.ndarray` should be used

Reply:
Yes, the wiki should be updated to show ``` :class:`numpy.ndarray` ```. I'll make that correction momentarily,

Reply:
> Yes, it used to be `Colors` and not clickable prior to my recommendation/suggestion: [#1064 (comment)](https://github.com/ManimCommunity/manim/pull/1064#discussion_r584241033)
> 
> @ksachdeva which addon is this? I wonder if it's configurable to automatically add this `~.` prefix to files from the local Manim library.

addon -

https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring